src/algebraic_geometry/open_immersion.lean

```diff
@@ -274,6 +280,243 @@ end
+      induction V using opposite.rec,
+      simp only [quiver.hom.unop_op, Top.presheaf.pushforward_obj_map, category.assoc,
+        nat_trans.naturality_assoc, functor.op_map, inv_naturality_assoc, ← Y.presheaf.map_comp],
+      erw ← Y.presheaf.map_comp,
+      congr
+    end } }
+
+lemma pullback_cone_of_left_condition :
+  pullback_cone_of_left_fst f g ≫ f = Y.of_restrict _ ≫ g :=
+begin
+  ext U,
+  { induction U using opposite.rec,
+    dsimp only [comp_c_app, nat_trans.comp_app, unop_op,
+      whisker_right_app, pullback_cone_of_left_fst],
+    simp only [quiver.hom.unop_op, Top.presheaf.pushforward_obj_map, app_inv_app_assoc,
+      eq_to_hom_app, eq_to_hom_unop, category.assoc, nat_trans.naturality_assoc, functor.op_map],
+    erw [← Y.presheaf.map_comp, ← Y.presheaf.map_comp],
+    congr },
+  { simpa using pullback.condition }
+end
+
+/--
+We construct the pullback along an open immersion via restricting along the pullback of the
+maps of underlying spaces (which is also an open embedding).
+-/
+def pullback_cone_of_left : pullback_cone f g :=
+pullback_cone.mk (pullback_cone_of_left_fst f g) (Y.of_restrict _)
+  (pullback_cone_of_left_condition f g)
+
+variable (s : pullback_cone f g)
+
+/--
+  (Implementation.) Any cone over `cospan f g` indeed factors through the constructed cone.
+-/
+def pullback_cone_of_left_lift : s.X ⟶ (pullback_cone_of_left f g).X :=
+{ base := pullback.lift s.fst.base s.snd.base
+  (congr_arg (λ x, PresheafedSpace.hom.base x) s.condition),
+  c :=
+  { app := λ U, s.snd.c.app _ ≫ s.X.presheaf.map (eq_to_hom (begin
+      dsimp only [opens.map, is_open_map.functor, functor.op],
+      congr' 2,
+      let s' : pullback_cone f.base g.base := pullback_cone.mk s.fst.base s.snd.base _,
+      have : _ = s.snd.base := limit.lift_π s' walking_cospan.right,
+      conv_lhs { erw ← this, rw coe_comp, erw ← set.preimage_preimage },
+      erw set.preimage_image_eq _
+        (Top.snd_open_embedding_of_left_open_embedding hf.base_open g.base).inj,
+      simp,
+    end)),
+    naturality' := λ U V i,
+    begin
+      erw s.snd.c.naturality_assoc,
+      rw category.assoc,
+      erw [← s.X.presheaf.map_comp, ← s.X.presheaf.map_comp],
+      congr
+    end } }
+
+lemma pullback_cone_of_left_lift_fst :
+  pullback_cone_of_left_lift f g s ≫
+    (pullback_cone_of_left f g).fst = s.fst :=
+begin
+  ext x,
+  { induction x using opposite.rec,
+    change ((_ ≫ _) ≫ _ ≫ _) ≫ _ = _,
+    simp_rw [category.assoc],
+    erw ← s.X.presheaf.map_comp,
+    erw s.snd.c.naturality_assoc,
+    have := congr_app s.condition (op (hf.open_functor.obj x)),
+    dsimp only [comp_c_app, unop_op] at this,
+    rw ← is_iso.comp_inv_eq at this,
+    reassoc! this,
+    erw [← this, hf.inv_app_app_assoc, s.fst.c.naturality_assoc],
+    simpa },
+  { change pullback.lift _ _ _ ≫ pullback.fst = _,
+    simp }
+end
+
+lemma pullback_cone_of_left_lift_snd :
+  pullback_cone_of_left_lift f g s ≫
+    (pullback_cone_of_left f g).snd = s.snd :=
```

Comment:
```suggestion
  pullback_cone_of_left_lift f g s ≫ (pullback_cone_of_left f g).snd = s.snd :=
```
Should this be a simp-lemma?

Reply:
I don't think so. I don't think one would ever want to use this construction outside of this file.